src/scripts/main.js

```diff
@@ -1,3 +1,230 @@
 'use strict';
 
-// write your code here
+let board;
+let score = 0;
+const rows = 4;
+const columns = 4;
+
+const button = document.querySelector('.start');
+
+button.addEventListener('click', () => {
+  document.querySelector('#board').innerHTML = '';
+  setGame();
+  setTwo();
+  setTwo();
+
+  document.querySelector('.message-start').classList.add('hidden');
+
+  button.classList.remove('start');
+  button.classList.add('restart');
+  button.innerText = 'Restart';
+});
+
+window.onload = function() {
+  setGame();
+};
+
+function setGame() {
+  board = [
+    [0, 0, 0, 0],
+    [0, 0, 0, 0],
+    [0, 0, 0, 0],
+    [0, 0, 0, 0],
+  ];
+
+  for (let r = 0; r < rows; r++) {
+    for (let c = 0; c < columns; c++) {
+      const tile = document.createElement('div');
+
+      tile.id = r.toString() + '-' + c.toString();
+
+      const num = board[r][c];
+
+      updateTile(tile, num);
+      document.getElementById('board').append(tile);
+    }
+  }
+}
+
+function hasEmptyTile() {
+  for (let r = 0; r < rows; r++) {
+    for (let c = 0; c < columns; c++) {
+      if (board[r][c] === 0) {
+        return true;
+      }
+    }
+  }
+
+  return false;
+}
+
+function setTwo() {
+  if (!hasEmptyTile()) {
+    document.querySelector('.message-lose').classList.remove('hidden');
+
+    return;
+  }
+
+  let found = false;
+
+  while (!found) {
+    // random r, c
+    const r = Math.floor(Math.random() * rows); // 0-1 * 4 -> 0 - 4
+    const c = Math.floor(Math.random() * columns);
+
+    if (board[r][c] === 0) {
+      const randomNums = [2, 2, 2, 2, 2, 2, 2, 2, 2, 4];
+
+      board[r][c] = randomNums[Math.floor(Math.random() * 10)];
+
+      const tile = document.getElementById(r.toString() + '-' + c.toString());
+
+      tile.innerText = board[r][c];
+      tile.classList.add('x' + board[r][c].toString());
+      found = true;
+    }
+  }
+}
+
+function updateTile(tile, num) {
+  tile.innerText = '';
+  tile.classList.value = ''; // clear the classList
+  tile.classList.add('tile');
+
+  if (num > 0) {
+    tile.innerText = num;
+    tile.classList.add('x' + num.toString());
+  }
+}
+
+document.addEventListener('keyup', (e) => {
+  if (e.code === 'ArrowLeft' && button.classList.contains('restart')) {
+    slideLeft();
+    setTwo();
+  } else if (e.code === 'ArrowRight' && button.classList.contains('restart')) {
+    slideRight();
+    setTwo();
+  } else if (e.code === 'ArrowUp' && button.classList.contains('restart')) {
+    slideUp();
+    setTwo();
+  } else if (e.code === 'ArrowDown' && button.classList.contains('restart')) {
+    slideDown();
+    setTwo();
+  }
+
+  document.querySelector('.score').innerText = score;
+});
+
+function filterZero(argRow) {
+  return argRow.filter(num => num !== 0); // create a new array without zeroes
```

Comment:
optional | if you really need get all elements except for 0 - it's ok. But now it looks like it could be shorten
```suggestion
  return argRow.filter(num => num); // create a new array without zeroes
```